portfolio/src/main/webapp/script.js

```diff
@@ -91,3 +91,8 @@ function changeImageBorders(color){
     document.images.item(i).style.border = "10px solid " + color;
   }
 }
+
+function sayHi() {
+  fetch('/data').then(response => response.text()).then(
```

Comment:
Nit: My preference would be to do this work (response.text()) when setting the element text, not as a separate step here. As is, it looks like response.text() returns a promise (though javascript supports both promise returning and value returning functions in `then()`).